lib/backends/memcached.js

```diff
@@ -32,46 +32,86 @@
 
 'use strict';
 
-const promisify = require('bluebird').promisify;
+const { promisify } = require('util');
 const Memcached = require('memcached');
 
-const noop = () => undefined;
+/**
+ * @typedef {Object} MemcachedOpts
+ * @property {Memcached.Location} hosts
+ * @property {Memcached} [client]
+ * @link https://github.com/3rd-Eden/memcached#options
+ */
 
+/**
+ * @param {MemcachedOpts & Memcached.config} options
+ */
 function createClient(options) {
-  if (options.client) {
+  if (options.client instanceof Memcached) {
     return options.client;
   }
   const hosts = options.hosts || '127.0.0.1:11211';
   return new Memcached(hosts, options);
 }
 
+/**
+ * @param {any} value
+ * @return {null|any}
```

Comment:
```suggestion
* @template T
 * @param {T} value
 * @return {null|T}
```
I think that's right (but double-check it)